### DISCUSSION TITLE
Setup PowerModelParam

### DISCUSSION THREAD

**Discussion by steven918** [gem5/gem5#1798] [Q&A]
Hi,

This maybe obvious but it's not so after trying. I can see there is a power_model parameter in the ClockedObjectParams:

`namespace gem5
{
struct ClockedObjectParams
    : public SimObjectParams
{
    gem5::ClockDomain * clk_domain;
    std::vector< gem5::PowerModel * > power_model;
    gem5::PowerState * power_state;
};

} // namespace gem5`

How to setup a power_model instance? For example, if assuming a clk_domain object in a system, which is derived from SrcClockDomain, then I would think the following should work:

`system.clk_domain.power_model = PowerModel()`

But there was an unproxy error:

"AttributeError: Can't resolve proxy 'any' of type 'SubSystem" from 'system.clk_domain.power_model'.  

Any suggestion?

Thanks.

**Comment by powerjg** [comment]
https://bootcamp.gem5.org/#02-Using-gem5/10-modeling-power may help.

**Reply by powerjg** [reply]
There may also be a video on YouTube (@gem5)

**Comment by steven918** [comment]
Thanks Jason,

That link helps to see an example. A follow-up question on Page 8 of the slide: why add_power-model() needs to be called before board._preinstantiate? Does that mean the power model can't be instantiated without a board but just a simple system in which ClockedObjects sit in like L3Cache in the example, or anything else like a clock source. 

I did try the example, but with similar error like "Can't resolve proxy 'any' of type 'SubSystem' from 'system.cache.power_model" (I used a cache, not L3 to try out). I think the error may come from the board._pre_instantiate limitation?

Thanks
